src/hvm.rs

```diff
@@ -982,6 +983,61 @@ impl FromStr for Name {
 impl U120 {
   pub const ZERO: U120 = U120(0);
   pub const MAX: U120 = U120(2_u128.pow(120) - 1);
+
+  pub fn wrapping_add(self, other:U120) -> U120 {
+    let res = self.0 + other.0;
+    U120(res & U120::MAX.0)
+  }
+  
+  pub fn wrapping_sub(self, other: U120) -> U120 {
+    let other_complement = U120::wrapping_add(U120(other.0 ^ U120::MAX.0), U120(1));
+    U120::wrapping_add(self, other_complement)
+  }
+
+  // based off of this answer https://stackoverflow.com/a/1815371
+  // maybe this is too much work for an easy function?
+  // idk, maybe there's a better way to do this
+  pub fn wrapping_mul(self, other: U120) -> U120 {
+    const LO_MASK : u128  =  (1 << 60) - 1;
+    let a = self.0;
+    let b = other.0;
+    let x = (a & LO_MASK) * (b & LO_MASK);
+    
+    let s0 = x & LO_MASK;
+    let x = (a >> 60) * (b & LO_MASK) + (x >> 60);
+    
+    let s1 = x & LO_MASK;
+    let s2 = x >> 60;
+    
+    let x = s2 + (a >> 60) * (b >> 60) + (x >> 60);
+    U120(s1 << 60 | s0)
+  }
+
+  // Wrapping div is just normal division, since
+  // self / other is always smaller than self.
+  // warning: this will panic when other is 0.
+  pub fn wrapping_div(self, other: U120) -> U120 {
+    U120(self.0 / other.0)
+  }
+
+  // Wrapping remainder is just normal remainder
+  // given that self % other is always smaller than other
+  // by definition of the other.
+  pub fn wrapping_rem(self, other: U120) -> U120 {
+    U120(self.0 % other.0)
+  }
+
+  // Wrapping shift left is only defined for
+  // values `other` between 0 and 127. For values bigger than
+  // that, it will wrap the value module 120 before doing the shift.
```

Comment:
Why wrap `other` to 120 instead of shifting `self` to 0 like the normal definition of << and >>?

Reply:
C standard states that it [shifting a number by more than it's width is undefined behaviour](https://wiki.sei.cmu.edu/confluence/display/c/INT34-C.+Do+not+shift+an+expression+by+a+negative+number+of+bits+or+by+greater+than+or+equal+to+the+number+of+bits+that+exist+in+the+operand). The way rust circumvents this is by defining `wrapping_shl`, which will be only defined in that range and the other values are taken modulo the width (javascript does the same thing i think). Based off of that, I implemented the same semantic for shl, and it would be perfectly valid (though less meaningful) to return zero when those cases return. I think it just comes down to our standards and semantics (and I would argue that there's a point doing it like this).

Reply:
@developedby  The question is: the normal definition?

Reply:
Normal for me would be to shift into oblivion.
```
0001 << 0 = 0001
0001 << 1 = 0010
0001 << 2 = 0100
0001 << 3 = 1000
0001 << 4 = 0000
```
Same for right shift.

This way, `(a << b) << c = a << (b + c)` which is how i personally intuitively think about it.

Reply:
But what language does that?

On C `0001 << 4` is undefined behaviour. On JS `2 << 32 == 2`.

Reply:
I guess I never actually did that (because why would you)